auto-api/src/main/java/io/opentelemetry/instrumentation/auto/api/WeakMap.java

```diff
@@ -87,10 +87,14 @@ public static boolean isProviderRegistered() {
   }
 
   class MapAdapter<K, V> implements WeakMap<K, V> {
+    private final Object[] locks = new Object[16];
```

Comment:
Doesn't have to block this PR, but do we ever use the `synchronizedMap` fallback? If we're using `WeakConcurrentMap` as a general default, or a Guava concurrent map otherwise, then we could remove the locks and `computeIfAbsent` (only used once in Netty instrumentation, `putIfAbsent` which is present on `WeakConcurrentMap` should work ok for it) and simplify things a lot.

Reply:
I recommend dropping this commit. The change here was made for symmetry with the other class Adapter and doesn’t really make much sense given that this one only wraps the synchronised wrapper. I also expect that we will simplify/refactor this soon, so if you keep syncing changes this would lead to some churn.